File: app/src/main/java/me/pwcong/jpstart/mvp/model/MemoryFragmentModelImpl.java
package me.pwcong.jpstart.mvp.model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import me.pwcong.jpstart.manager.DBManager;
import me.pwcong.jpstart.mvp.bean.JPItem;

/**
 * Created by Pwcong on 2016/10/5.
 */

public class MemoryFragmentModelImpl implements BaseModel.MemoryFragmentModel {


    @Override
    public List<JPItem> getQingYinWithoutHeader() {

        List<JPItem> list = new ArrayList<>();
        list.addAll(DBManager.getInstance().getQingYinWithoutHeader());

        Collections.shuffle(list);

        return list;

    }

    @Override
    public List<JPItem> getZhuoYinWithoutHeader() {

        List<JPItem> list = new ArrayList<>();
        list.addAll(DBManager.getInstance().getZhuoYinWithoutHeader());

        Collections.shuffle(list);

        return list;
    }

    @Override
    public List<JPItem> getAoYinWithoutHeader() {

        List<JPItem> list = new ArrayList<>();
        list.addAll(DBManager.getInstance().getAoYinWithoutHeader());

        Collections.shuffle(list);

        return list;
    }
}

File: app/src/main/java/com/github/pwcong/jpstart/mvp/model/impl/MemoryFragmentModelImpl.kt
package com.github.pwcong.jpstart.mvp.model.impl

import com.github.pwcong.jpstart.manager.DBManager
import com.github.pwcong.jpstart.mvp.bean.JPItem
import com.github.pwcong.jpstart.mvp.model.BaseModel.MemoryFragmentModel

class MemoryFragmentModelImpl : MemoryFragmentModel {
    override val qingYinWithoutHeader: List<JPItem>
        get() {
            val list: MutableList<JPItem> = mutableListOf()
            list.addAll(DBManager.getInstance().getQingYinWithoutHeader())
            list.shuffle()

            return list
        }

    override val zhuoYinWithoutHeader: List<JPItem>
        get() {
            val list: MutableList<JPItem> = mutableListOf()
            list.addAll(DBManager.getInstance().getZhuoYinWithoutHeader())
            list.shuffle()

            return list
        }

    override val aoYinWithoutHeader: List<JPItem>
        get() {
            val list: MutableList<JPItem> = mutableListOf()
            list.addAll(DBManager.getInstance().getAoYinWithoutHeader())
            list.shuffle()

            return list
        }
}
